### PR TITLE
[s] Fixes ore redemption exploit

### LINKED ISSUE #35057: You can duplicate bluespace crystals
Issue reported from  Round ID: 82693 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

-Upgrade the ORM with a better matter bin
-Smelt a bluespace crystal
-Take out the bluespace mesh from the ORM
-With the mesh in one hand, use an empty hand on it to pick off a piece of refined crystal
-Smelt the refined crystal 

You now have more bluespace than you started with. Repeat infinitely.
ex4

This doesn't get you more points, just more bluespace.

### PR DESCRIPTION
**Pull request by Poojawa**
Since the crystal is already refined, it would make sense for the refinery to reject it. 

fixes #35057

### CODE REVIEW

**Comment by Barhandar** [comment]
I can bet it just makes it impossible to _smelt the "ore" from lavaland_ instead of preventing banach-tarskiing it.

Actual fix would be, you know, not having the thing you need to stick into machinery be exactly same item as the ore you get.

**Comment by Poojawa** [comment]
Uh, no. If you looked at the bluespace crystal ore code, you'll see that refined is a subtype of the ore. Thus with no checks the machine just assumed it was the same thing and accepted it as 'new' ore entirely. 

Artificial has `refined_type = null`

**Comment by Poojawa** [comment]
Course, having just tested it, you can insert artificial ones in, but since they take diamond to make, I think it's fair.

File: code/modules/mining/machine_redemption.dm [hunk 1]
```diff
@@ -52,6 +52,9 @@
 	var/datum/component/material_container/mat_container = materials.mat_container
 	if (!mat_container)
 		return
+		
+	if(istype(O, /obj/item/stack/ore/bluespace_crystal/refined))
```
Comment (AnturK) [inline code/modules/mining/machine_redemption.dm]: Just do a generic refined_type == null check here ?

**Review by AnturK — CHANGES_REQUESTED** [review]
Generalize this please.

File: code/modules/mining/machine_redemption.dm (hunk 1 above)
Comment (Poojawa) [reply code/modules/mining/machine_redemption.dm]: Easy enough

File: code/modules/mining/machine_redemption.dm [hunk 2]
```diff
@@ -179,6 +182,14 @@
 		if(user.transferItemToLoc(W, src))
 			inserted_disk = W
 			return TRUE
+			
+	if(istype(W, /obj/item/stack/ore))
```
Comment (AnturK) [inline code/modules/mining/machine_redemption.dm]: Now you have the same istype done twice.

Comment (Poojawa) [reply code/modules/mining/machine_redemption.dm]: yes, it error'd out as just W, specifying it again passed compile check locally.  

`code/modules/mining/machine_redemption.dm:187:error: W.refined_type: undefined var`

Comment (Poojawa) [reply code/modules/mining/machine_redemption.dm]: as an addendum, it was attack_hand that was ignoring the check before as well, if you throw a refined crystal on the floor, it gets recycled with no additions, but if you click on it, you bypass the auto grab check, this is why I added it here as well.

Comment (AnturK) [reply code/modules/mining/machine_redemption.dm]: You can just do
```
var/obj/item/stack/ore/O = W
if(istype(O)) do_stuff
```

**Pull request merged**